dfastmi/gui/dialog_view_model.py

```diff
@@ -110,10 +112,32 @@ def current_reach(self, value):
         self._current_reach = value
         self._initialize_qthreshold()
         self._initialize_ucritical()
-        self._update_qvalues()
         # Notify the view of the change
         self.reach_changed.emit(self._current_reach.name)
 
+    @property
+    def qthreshold(self) -> float:
+        """
+        The current threshold discharge.
+        """
+        return self._qthreshold
+
+    @qthreshold.setter
+    def qthreshold(self, value: float):
+        """
+        Setter for the current threshold discharge.
+        After set notify the view of the change.
+
+        Arguments:
+            value (float): The threshold discharge to set.
+        """
+        self._qthreshold = value
```

Comment:
Please check if new qthreshhold is already same as the one currently in view_model, no need to emit again.
```
if self_qthreshold == value:
    return
    
```

Reply:
Discussed with Bert.